saleor/order/calculations.py

```diff
@@ -525,6 +525,65 @@ def order_line_tax_rate(
     return order_line.tax_rate
 
 
+def order_line_unit_discount(
+    order: Order,
+    order_line: OrderLine,
+    manager: PluginsManager,
+    lines: Optional[Iterable[OrderLine]] = None,
+    force_update: bool = False,
+) -> Decimal:
+    """Return the line unit discount.
+
+        It takes into account all plugins.
+        If the prices are expired, call all order price calculation methods
+        and save them in the model directly.
+    g
```

Comment:
Same comments as for: https://github.com/saleor/saleor/pull/16218#pullrequestreview-2145012256